metrics-exporter/src/bin/io_engine/client/grpc_client.rs

```diff
@@ -167,7 +114,49 @@ pub(crate) async fn init_client(api_version: ApiVersion) -> Result<GrpcClient, E
         .path_and_query("")
         .build()
         .map_err(|error| ExporterError::InvalidURI(error.to_string()))?;
-    let ctx = GrpcContext::new(endpoint, timeout, api_version);
+    let ctx = GrpcContext::new(endpoint, timeout);
     let client = GrpcClient::new(ctx).await?;
     Ok(client)
 }
+
+impl GrpcClient {
+    /// Gets Capacity statistics of all pool on the io engine.
+    /// Maps the response to PoolInfo struct.
+    pub(crate) async fn list_pools(&self) -> Result<Pools, ExporterError> {
+        let pools = match self
+            .client_v1()?
+            .pool
+            .list_pools(rpc::v1::pool::ListPoolOptions::default())
+            .await
+        {
+            Ok(response) => response
+                .into_inner()
+                .pools
+                .into_iter()
+                .map(PoolInfo::from)
+                .collect::<Vec<_>>(),
+            Err(error) => return Err(ExporterError::GrpcResponseError(error.to_string())),
+        };
```

Comment:
```suggestion
            Ok(response) => Ok(response
                .into_inner()
                .pools
                .into_iter()
                .map(PoolInfo::from)
                .collect::<Vec<_>>()),
            Err(error) => Err(ExporterError::GrpcResponseError(error.to_string())),
        }?;
```

Reply:
We return below this. We will return `Ok(PoolIoStats { pool_stats })`. pool_stats is Vec<PoolInfo>

Reply:
Yes I see that, my point is that don't need to explicitly return, can use ? instead which is more idiomatic

Reply:
Done